judge/judgedaemon.main.php

```diff
@@ -333,7 +333,8 @@ function fetch_executable_internal(
     $execbuildpath   = $execbuilddir . '/build';
     $execrunpath     = $execbuilddir . '/run';
     $execrunjurypath = $execbuilddir . '/runjury';
-    if (!is_dir($execdir) || !file_exists($execdeploypath)) {
+    if (!is_dir($execdir) || !file_exists($execdeploypath) ||
+        ($combined_run_compare && file_get_contents(LIBJUDGEDIR . '/run-interactive.sh')!==file_get_contents($execrunpath))) {
```

Comment:
Why are these additional checks required?

Reply:
Yes, unrelated, but both happen due to upgrading a system with existing data.